tests/vcs/git/test_backend.py

```diff
@@ -41,3 +47,25 @@ def test_invalid_revision_sha_max_len() -> None:
 def test_get_name_from_source_url(url: str) -> None:
     name = Git.get_name_from_source_url(url)
     assert name == "poetry"
+
+
+def test_annotated_tag() -> None:
```

Comment:
Can you make the test `parametrized` to test passing `bytes` as well as `str`, please?

Reply:
CMMIR, 
```
    @staticmethod
    def get_name_from_source_url(url: str) -> str:
        return re.sub(r"(.git)?$", "", url.rstrip("/").rsplit("/", 1)[-1])
```
 Does not accept bytes as url right? So passing bytes as url will fail unless we change this implementation.